src/Resources/config/app/services/factory.xml

```diff
@@ -54,5 +54,19 @@
             <argument type="service" id="brille24.repository.customer_option_value"/>
             <argument type="service" id="brille24.customer_options_plugin.factory.customer_option_value_price_factory" />
         </service>
+
+        <service
+                class="Brille24\SyliusCustomerOptionsPlugin\Factory\CartItemFactory"
+                id="sylius.custom_factory.order_item"
```

Comment:
We have a naming schema. `brille24.factory.order_item` would probably fit better.

Reply:
IMHO I need the Sylius ID here to overwrite the service definition. Other services rely on that ID, that's why I picked it. Or am I missing something?

Reply:
@shochdoerfer this is a decorator. You do not need to overwrite the service ID. The decorator gets its own id. And when the service it decorates is used, the decorator is applied automatically. Or symfony replaces the old one with the decorated one automatically? Anyway, looking at the docs and recalling my experience with decorators, I am certain you do not overwrite the service ID in the config. You just need to configure what service you intend to decorate.
Have you tried using the id @mamazu suggested? I think that should work just fine.

https://symfony.com/doc/6.4/service_container/service_decoration.html

Reply:
@seizan8 that makes sense. Can't recall why it did not work for me or it looked like it is not working. 

I've fixed the issue locally already. Currently checking how to best fix the tests as things changed quite a bit.